packages/machine-extractor/src/extractAction.ts

```diff
@@ -189,56 +152,43 @@ export function extractLogAction(
 export function extractStopAction(
   actionNode: ActionNode,
   fileContent: string,
-): { type: 'string' | 'expression'; value: string } {
+): string | JsonExpressionString {
   const node = actionNode.node;
   if (t.isCallExpression(node)) {
     const arg = node.arguments[0];
 
     // stop('id')
     if (t.isStringLiteral(arg)) {
-      return { type: 'string', value: arg.value };
+      return arg.value;
     }
 
     // stop(() => {}) or anything else
-    return {
-      type: 'expression',
-      value: fileContent.slice(arg.start!, arg.end!),
-    };
+    return `{{${fileContent.slice(arg.start!, arg.end!)}}}`;
   }
 
   throw Error('Unsupported stop action');
 }
 
-type SendToEventArg =
-  | { type: 'expression'; value: string }
-  | {
-      type: 'object';
-      value: Record<string, JsonEntry>;
-    }
-  | undefined;
-type SendToActorRefArg =
-  | { type: 'string' | 'expression'; value: string }
-  | undefined;
-type SendToOptionArg =
-  | { type: 'string' | 'expression'; value: string }
-  | { type: 'number'; value: number }
-  | undefined;
+type SendToEventArg = JsonExpressionString | Record<string, JsonItem>;
+type SendToActorRefArg = string | JsonExpressionString;
+type SendToIdArg = string | JsonExpressionString;
+type SendToDelayArg = string | number | JsonExpressionString;
 
 export function extractSendToAction(
   actionNode: ActionNode,
   fileContent: string,
 ): {
   event: SendToEventArg;
   to: SendToActorRefArg;
-  delay: SendToOptionArg;
-  id: SendToOptionArg;
+  delay: SendToDelayArg;
+  id: SendToIdArg;
 } {
   const node = actionNode.node;
-  let eventObject: SendToEventArg = undefined;
-  let actorRef: SendToActorRefArg = undefined;
-  const options: { [key: string]: SendToOptionArg } = {
-    delay: undefined,
-    id: undefined,
+  let eventObject: SendToEventArg = {};
+  let actorRef: SendToActorRefArg = '';
+  const options: { id: SendToIdArg; delay: SendToDelayArg } = {
+    id: '',
+    delay: 0,
```

Comment:
This shouldn't use any defaults. What is not provided in the source code should stay as not provided, especially since there is a difference between `delay: 0` and `delay: undefined`. The first one instructs the runtime to use `setTimeout` whereas the second one doesn't (the event gets sent immediately/synchronously)

Reply:
Thanks for catching this. I removed those defaults in the other PR.